File: greedy algorithm/coins/coins.cpp
#include <bits/stdc++.h>
#include <iostream>
using namespace std;

int solve_change(int change, vector<int> coins, int amount_coins, vector<int> &solution);

int main(){

    vector<int> coins = {100, 50, 25, 10, 5, 1};
    int amount_coins = (int) coins.size();

    int change = 0;

    cout << "What is the change? (in cents): ";
    cin >> change;

    vector<int> solution;
    
    int solution_exist = solve_change(change, coins, amount_coins, solution);

    if(solution_exist){
        cout << "A possible solution is: ";
        for( int i = 0; i < (int) solution.size(); i++) cout << solution[i] << " ";
    }
    else
        cout << "There's no possible solution.\n";

    return 0;
}

int solve_change(int change, vector<int> coins, int amount_coins, vector<int> &solution){
    int sum = 0, counter = 0;

    while (sum != change){
        if(sum + coins[counter] <= change){
            solution.push_back(coins[counter]);
            sum += coins[counter];
        }
        else if(counter == coins.size())
            return 0;
        else
            counter += 1;
    }

    return 1;
}

File: greedy algorithm/coins/coins.py
def change(N):
    cents = [100, 50, 25, 10, 5, 1]
    solution = []
    sum = 0

    i = 0   #find coin position in array
    while sum != N:
        if sum + cents[i] <= N:
            x = cents[i]
            sum += cents[i]
            solution.append(cents[i]) 
        elif i == len(cents):
            return 0
        else:
            i += 1
    return solution

if __name__ == "__main__":
    n = int(input("Wha't is the change? (in cents) "))
    solution = change(n)
    
    if solution: print(solution)
    else: print("No solution found")